libexec/bats-exec-test

```diff
@@ -89,11 +90,16 @@ bats_test_begin() {
 
 bats_test_function() {
   local test_name="$1"
-  BATS_TEST_NAMES["${#BATS_TEST_NAMES[@]}"]="$test_name"
+  BATS_TEST_NAMES+=("$test_name")
 }
 
+BATS_CURRENT_STACK_TRACE=()
+BATS_PREVIOUS_STACK_TRACE=()
+
 bats_capture_stack_trace() {
-  BATS_PREVIOUS_STACK_TRACE=( "${BATS_CURRENT_STACK_TRACE[@]}" )
+  if [[ "${#BATS_CURRENT_STACK_TRACE[@]}" -ne '0' ]]; then
```

Comment:
Nit: not that it makes a difference to parsing, but `'0'` -> integer `0` ?

Reply:
It's a preference thing: I've been bitten by unquoted values so many times, that rather than try to remember when to quote/when not to, I literally quote _everything_. (Except for variables used as the pattern in a `=~` expression, which you never want to quote.)

Granted, it may seem like overkill for integer comparisons, and I'd eventually relent on this point if the pressure mounts; but for now I'd just as soon leave it.

Reply:
No pressure at all, nonstandard but for good reason.

Perhaps we could have a project styleguide? I default to https://google.github.io/styleguide/shell.xml but anything would do. Will open an issue if there's nothing already.

Reply:
Your detailed styleguide at https://github.com/mbland/go-script-bash/blob/master/CONTRIBUTING.md#formatting is great @mbland!

Reply:
Aww shucks, thanks! Feel free to open an issue and maybe reference mbland/go-script-bash#29. I've been wanting to do something like that for a long time.

Not to oversell my wares, but I was considering using [mbland/guides-template](https://github.com/mbland/guides-template) to produce it in the near future, as I'm presently dusting it off for use at my new company. I have a whole theory behind why I want to use it; my very rough [Documenation as organization hacking tools](https://mike-bland.com/docs-as-hacking-tools) deck contains a bit about it.

Also, regarding the Google style guide: Full disclosure, I'm an ex-Googler. We didn't have the Bash guide when I was there (2005-2011), I don't think—but when I came across it a few weeks ago, I was surprised/not surprised to find I wrote most of mbland/go-script-bash in conformance with it already! 😛 

This speaks to the power of good conventions that transcend individual languages, and guides are critical to developing and disseminating them. This said, I'm a proponent of using existing work as a _template_ (see what I did there?) and fitting it to the project/organization as needed. If innovations emerge, then they can be proposed for inclusion upstream. (And if only we had a tool that supported such a distributed editing model...) I find it provides a nice balance between leveraging prior art and allowing for shared ownership rather than slavish insistence on conformity for conformity's sake.

Enough philosophy for one code review comment? 😛

Reply:
> slavish insistence on conformity for conformity's sake
Ah yes, oft observed.

Agree strongly on the contents of that slide deck, and support using https://github.com/mbland/guides-template - thanks for the time and thought you're contributing to this project/ecosystem :smile: Bring on the philosophy!

Reply:
I'm also having a visceral reaction to the quoted integer... :)

If we are going to quote it (personally, I'm still 👎  on quoting literals, especially integers), can we at least stick with doublequotes?